firefox_puppeteer/tests/test_about_window.py

```diff
@@ -0,0 +1,63 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+from marionette_driver import By
+
+from firefox_ui_harness.testcase import FirefoxTestCase
+
+
+class TestAboutWindow(FirefoxTestCase):
+
+    def setUp(self):
+        FirefoxTestCase.setUp(self)
+
+        self.about_window = self.browser.open_about_window(trigger='menu')
+
+    def tearDown(self):
+        try:
+            self.windows.close_all([self.browser])
+        finally:
+            FirefoxTestCase.tearDown(self)
+
+    def test_basic(self):
+        self.assertEqual(self.about_window.window_type, 'Browser:About')
+
+    def test_check_for_updates_and_download(self):
+        # testing both of these as they need to occur in this order
+        self.about_window.check_for_updates()
+        self.assertEqual('downloadAndInstall', self.about_window.wizard_state)
+        self.about_window.download()
+        self.assertEqual('apply', self.about_window.wizard_state)
+
+    def test_close_window(self):
+        """Test closing the About window."""
+        self.about_window.close()
+        self.assertTrue(self.about_window.closed)
+
+    def test_open_window(self):
+        """Test various opening strategies."""
+        def opener(win):
+            menu = win.marionette.find_element(By.ID, 'aboutName')
+            menu.click()
+
+        open_strategies = ('menu',
+                           opener,
+                           )
+
+        for trigger in open_strategies:
+            if not self.about_window.closed:
+                self.about_window.close()
```

Comment:
This if condition should never be necessary in this loop. Given that you open a new about window in setUp() you want to call `close()` before the for loop.